### PR TITLE
Stop asserting integration test run time & move to performance testing runner.

### PR DESCRIPTION
**Pull request by michaelkaye**
Performance tests like these are not best run on a 100% success rate. We should not assert a failure due to a timeout and instead capture the range of times that the test runs in.

A single failure means nothing; we need to look at patterns instead, and allow ourselves to work around networking glitches or cpu spikes on the test server.

### CODE REVIEW

**Comment by michaelkaye** [comment]
We also aim to persist the device logs as well as the test logs here.

**Comment by stefanceriu** [comment]
In my defense the values we assert on are averages over 3 runs (+ a warmup one)

**Comment by michaelkaye** [comment]
This now additionally moves the integration tests over to the performance testing GHA runner in the office.

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -2,14 +2,14 @@ name: Integration tests
 
 on:
   schedule:
-    - cron:  '0 2 * * 1-5'
```
Comment (michaelkaye) [inline .github/workflows/integration-tests.yml:5]: We own the hardware so can run this ... as often as we like ^^

**Comment by michaelkaye** [comment]
There were some unit test failures but i believe they're unrelated; have updated to latest develop and hopefully that should clear it up.

**Comment by michaelkaye** [comment]
https://github.com/vector-im/element-x-ios/actions/runs/5345012631 contains a demonstration of the integration tests running (against the branch).

https://vector-im.github.io/element-x-ios/performance/ shows the improvement in performance.

https://michaelkaye.github.io/element-x-ios/performance/ shows historical data (hourly for ~24h) is also more consistent over time.

**Review by Velin92 — APPROVED** [review]

**Pull request merged**